src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheFactory.java

```diff
@@ -0,0 +1,48 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import co.rsk.panic.PanicProcessor;
+import java.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PegoutSignedCacheFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger(PegoutSignedCacheFactory.class);
+  private static final PanicProcessor panicProcessor = new PanicProcessor();
+
+  private static PegoutSignedCache instance;
+
+  private PegoutSignedCacheFactory() {
+  }
+
+  /**
+   * Retrieves an instance of PegoutSignedCache with the specified Time To Live
+   * (TTL). If an instance does not already exist, it is lazily initialized.
+   * 
+   * @param ttl The duration for which the cache entries are considered valid.
+   * @return An instance of PegoutSignedCache.
+   * @throws IllegalArgumentException If the supplied TTL value is invalid.
+   */
+  public static PegoutSignedCache getInstance(final Duration ttl) throws IllegalArgumentException {
+    if (!isValidTtl(ttl)) {
```

Comment:
just as we discussed the last clean-code forum, we should avoid negative conditionals.
I'd suggest you call the method something like `ttlIsNotValid`

Reply:
very good one

Reply:
Sounds great, thank you both! I just updated the method to reflect what y'all described. My only contribution is to make the private method name `assertValidTtl`. Let me know what you think.